packages/gatsby/src/webpack/stats-extractor-plugin.js

```diff
@@ -0,0 +1,98 @@
+const fs = require(`fs-extra`)
+const path = require(`path`)
+const uniq = require(`lodash/uniq`)
+const chunkSorter = require(`./chunk-sorter`)
+
+const chunkOnlyConfig = {
+  assets: false,
+  cached: false,
+  children: false,
+  chunks: true,
+  chunkModules: false,
+  chunkOrigins: false,
+  errorDetails: false,
+  hash: false,
+  modules: false,
+  reasons: false,
+  source: false,
+  timings: false,
+  version: false,
+}
+
+module.exports = class GatsbyStatsExtractor {
+  constructor() {
+    this.statsPath = path.join(`public`, `webpack.stats.json`)
+    this.chunkMapPath = path.join(`public`, `chunk-map.json`)
+  }
+
+  sortChunks(stats) {
+    const { chunks } = stats.toJson(chunkOnlyConfig)
+    return chunkSorter(chunks, stats.compilation.chunkGroups)
+  }
+
+  apply(compiler) {
+    compiler.hooks.done.tapPromise(
```

Comment:
So this was `tapAsync` previously, now `tapPromise`?

I'm not a webpack guru, so not sure if it it's a meaningful difference 🤷‍♂️

Reply:
same thing, the only difference is you can use a promise instead of a callback, which is more inline with what we do elsewhere

Reply:
Didn't look closely enough to see if it was a refactor or a straight copy 🙃 Makes sense!